sqlmesh/core/schema_diff.py

```diff
@@ -294,15 +293,41 @@ class SchemaDiffer(PydanticModel):
     change. As a result historical data is lost.
 
     Potential future improvements:
-    1. Support precision changes on columns like VARCHAR and DECIMAL. Each engine has different rules on what is allowed
-    2. Support column moves. Databricks Delta supports moves and would allow exact matches.
+    1. Support column moves. Databricks Delta supports moves and would allow exact matches.
+
+    Args:
+        support_positional_add: Whether the engine for which the diff is being computed supports adding columns in a
+            specific position in the set of existing columns.
+        support_nested_operations: Whether the engine for which the diff is being computed supports modifications to
+            nested data types like STRUCTs and ARRAYs.
+        compatible_types: Types that are compatible and automatically coerced in actions like UNION ALL. Dict key is data
+            type, and value is the set of types that are compatible with it.
+        support_coercing_compatible_types: Whether or not the engine for which the diff is being computed supports direct
+            coercion of compatible types.
+        parameterized_type_defaults: Default values for parameterized data types. Dict key is a sqlglot exp.DataType.Type,
+            but in the engine adapter specification we build it from the dialect string instead of specifying it directly.
+            Example: `exp.DataType.build("STRING", dialect=DIALECT).this` instead of the underlying `exp.DataType.Type.TEXT`
+            to which it parses. We do that because parameter default replacement will silently break if we specify type
```

Comment:
i'm not sure that's a good reason for it, more indication that this should go into sqlglot